hasher-matcher-actioner/hmalib/lambdas/api/actions_api.py

```diff
@@ -71,17 +71,26 @@ def fetch_all_actions() -> FetchAllActionsResponse:
             actions_response=[config.__dict__ for config in action_configs]
         )
 
-    @actions_api.put("/", apply=[jsoninator(CreateUpdateActionRequest)])
-    def update_action(request: CreateUpdateActionRequest) -> UpdateActionResponse:
+    @actions_api.put(
+        "/<old_name>/<old_config_sub_stype>",
+        apply=[jsoninator(CreateUpdateActionRequest)],
+    )
+    def update_action(
+        request: CreateUpdateActionRequest, old_name: str, old_config_sub_stype: str
+    ) -> UpdateActionResponse:
         """
         Update an action url and headers
         """
-        config = ActionPerformer._get_subtypes_by_name()[request.config_subtype].getx(
-            request.name
-        )
-        for key, value in request.fields.items():
-            setattr(config, key, value)
-        hmaconfig.update_config(config)
+        if old_name != request.name or old_config_sub_stype != request.config_subtype:
```

Comment:
Maybe a comment about how these fields are immutable in our storage model so we must delete and create?